### ISSUE TITLE
supervised MNIST example accuacy and GPU issues

### ISSUE THREAD

**Issue by Thonner** [BindsNET/bindsnet#362]
Hi BindsNet
I am doing a master project where I am building a neuromorphic processor. I'm am (trying) to use your platform to train a network using Bindsnet, as this is the most promising SNN simulator that stays purely in the spiking domain. Long story to say that I have been playing around supervised MNIST examples, as I would like to use this to test my hardware implementation. But I'm seeing some issues.

If you want to use GPU it crashes in the `poisson` function in `encoding.py`. where it expects `cuda:0` device but gets `cpu`. Not sure but I think its the dataloader that needs to be sent to the right device.

Next: The accuracy measurements that I provided in the training loop can not be right. I have tested this the following way. First i train a network over 5000 inputs, using the provided training loop. This loop reports 100% accuracy after about 600 inputs. After training, I turn off training in the network. Then i test the accuacy after training with the following loop:
```python
pbar = tqdm(enumerate(dataloader))
    for (i, datum) in pbar:
        if i > 1000:
            break
        image = datum["encoded_image"]
        label = datum["label"]
        inputs = {"X": image.view(time, 1, 1, 28, 28)}
        network.run(inputs=inputs, time=time)

        out_spikes = network.monitors["Ae_spikes"].get("s")
        class_spike=torch.zeros(10)
        for i in range(time):
            for j in range(n_neurons):
                if out_spikes[i,0,j]:
                    class_spike[j//10] += 1
        
        aVal, maxInd = class_spike.max(0)
        if maxInd == label[0]:
            hit += 1

    
    acc = hit/1000
    print("\n accuracy: " + str(acc) +"\n")
```
This test returns accuracy around 65% depending on the seed. So what is the accuracy that the training loop is stating? To me, my loop should give the actual accuracy.

Sorry for the long issue! Your platform seems really cool! Therefore it is also a shame that your display piece is not running properly. I hope that you'll take the time to have a look at my complaints
- Anthon

**Comment by Hananel-Hazan** [comment]
Thank you for bringing this up. You are right, there was a couple of issues with the example that prevent it to work under GPU. 
The data loader has been fixed. 

The accuracy reported in the example is for the training process. It shows that the network reacts as expected. Once the training completes the supervise mechanism that clamps the right neuron and forces it to fire no longer valid. 
Thus in your code, the measurement accuracy is correct for the testing part, where there is no guidance from the supervisor. Feel free to adjust the network parameters (nu, theta_plus, etc) to achieve better accuracy. It's there, you just need to find it. 

About the plots, they indeed need to be improved, feel free to push a  PR, it will be gladly appreciated. 

Note that in your code you use 'i' twice, in the outer loop, and the inside loop.
